lib/src/validator/dependency.dart

```diff
@@ -223,7 +221,7 @@ class DependencyValidator extends Validator {
       constraint = '^${(dep.constraint as VersionRange).min}';
     } else {
       constraint = '"${dep.constraint} '
-          '<${(dep.constraint as VersionRange).min.nextBreaking}"';
+          '<${(dep.constraint as VersionRange).min!.nextBreaking}"';
```

Comment:
I wonder if we handle correctly the place where `dep.min` is null?
Perhaps we need to make another condition

Reply:
I think this might not be possible. We are in a situation where we are warning about a missing upper bound. We cannot both have a missing upper and a missing lower bound.

Reply:
I think the 'any' VersionRange has neither upper nor lower bound? https://github.com/dart-lang/pub_semver/blob/master/lib/src/version_constraint.dart#L19

Reply:
You are right.  I'll add a TODO, and then we can discuss what the right warning should be here.